src/main/java/io/reactivex/subjects/UnicastSubject.java

```
@@ -184,7 +191,7 @@ public void onNext(T t) {
             return;
         }
         if (t == null) {
-            onError(new NullPointerException());
+            onError(new NullPointerException("onNext called with null. Null values are generally not allowed in 2.x operators and sources."));
```

should not this throw immediately rather than trying to go through onError?

---

Yeah, the spec mandates throwing on null input and I think the Processor TCK also expects it. I thought signalling NPE is more graceful with Subjects and FlowableProcessors and shuts down the streams as well whereas a thrown NPE may leave everybody hanging.

---

Alright I'll leave the decision up to you. I'd prefer the fail early approach however delegating it down the stream also works. If you want me to change it I'll do so in a follow up PR.